src/main/java/seedu/address/logic/commands/UndoCommand.java

```diff
@@ -0,0 +1,98 @@
+package seedu.address.logic.commands;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+
+/**
+ * Represents an undo command in the address book.
+ * This undoes the last undo-able command, which includes edit, add, or delete commands.
+ * Optionally, you can specify the number of commands to undo.
+ */
+public class UndoCommand extends Command {
+    public static final String COMMAND_WORD = "undo";
+
+    public static final String COMMAND_FORMAT = "Command Format: " + COMMAND_WORD + " [number]\n"
+            + "Examples:\n"
+            + " - " + COMMAND_WORD + " (undoes the last command)\n"
+            + " - " + COMMAND_WORD + " 4 (undoes the last 4 commands)";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Undoes the last undo-able command. "
+            + "An undo-able command includes an edit, add, clear or delete command.\n"
+            + "Optionally, you can specify the number of commands to undo.\n"
+            + COMMAND_FORMAT;
+    public static final String INVALID_STEPS_TO_UNDO = "Undo step count cannot be a negative number or zero.\n";
+
+    public static final String INVALID_NATURAL_NUMBER_TO_UNDO = "Please provide a valid number of steps to undo, "
+            + "not exceeding the available command history";
+
+    public static final String NO_HISTORY_EXISTS_FAILURE = "There is no history of un-doable commands to be undone.\n"
+            + "Please execute some undo-able commands first.\n"
+            + "Undo-able commands include add, edit, delete or clear command.";
+
+    private static final Logger logger = Logger.getLogger(UndoCommand.class.getName());
+    private int stepsToUndo;
+
+
+    /**
+     * Constructs an UndoCommand for the specified number of undo steps.
+     *
+     * @param stepsToUndo The number of steps to undo.
+     */
+    public UndoCommand(int stepsToUndo) {
+        assert stepsToUndo > 0;
```

Comment:
I think that this assert might be unnecessary since the check for `stepsToUndo > 0` is already done in UndoCommandParser.

Reply:
On the contrary, I think it is good to make these assertions! Assertions are meant to ensure that some assumption is maintained, so having an assertion here can keep yourself from misusing methods.